src/GafferScene/TweakPlug.cpp

```diff
@@ -240,11 +279,127 @@ class NumericTweak
 			case TweakPlug::Multiply :
 				data->writable() = sourceDataCast->readable() * tweakDataCast->readable();
 				break;
+			case TweakPlug::Min :
+				/// \todo is this neccessary?
+				assert( sourceDataCast->readable().dimensions() == tweakDataCast->readable().dimensions() );
+
+				for( unsigned int i = 0; i < sourceDataCast->readable().dimensions(); ++i )
+				{
+					data->writable()[i] = std::min( sourceDataCast->readable()[i], tweakDataCast->readable()[i] );
+				}
+				break;
+			case TweakPlug::Max :
+				/// \todo is this neccessary?
+				assert( sourceDataCast->readable().dimensions() == tweakDataCast->readable().dimensions() );
+
+				for( unsigned int i = 0; i < sourceDataCast->readable().dimensions(); ++i )
+				{
+					data->writable()[i] = std::max( sourceDataCast->readable()[i], tweakDataCast->readable()[i] );
+				}
+				break;
+			case TweakPlug::ListAppend :
+			case TweakPlug::ListPrepend :
+			case TweakPlug::ListRemove :
+				throw IECore::Exception( boost::str( boost::format( "Cannot apply tweak with mode %s to \"%s\" : Data type %s not supported." ) % GafferScene::Detail::modeToString( m_mode ) % m_tweakName % m_sourceData->typeName() ) );
+			case TweakPlug::Replace :
+			case TweakPlug::Remove :
+			case TweakPlug::Create :
+				// These cases are unused - we handle them outside of numericTweak.
+				// But the compiler gets unhappy if we don't handle some cases
+				break;
+		}
+	}
+
+	void operator()( Data * data ) const
+	{
+		throw IECore::Exception( boost::str( boost::format( "Cannot apply tweak with mode %s to \"%s\" : Data type %s not supported." ) % GafferScene::Detail::modeToString( m_mode ) % m_tweakName % m_sourceData->typeName() ) );
+	}
+
+private:
+
+	const IECore::Data *m_sourceData;
+	const IECore::Data *m_tweakData;
+	TweakPlug::Mode m_mode;
+	const std::string &m_tweakName;
+};
+
+class ListTweak
+{
+public:
+	ListTweak( const IECore::Data *sourceData, const IECore::Data *tweakData, TweakPlug::Mode mode, const std::string &tweakName )
+		: m_sourceData( sourceData ), m_tweakData( tweakData ), m_mode( mode ), m_tweakName( tweakName )
+	{
+	}
+
+	template<typename T>
+	void operator()( T * data, typename std::enable_if<TypeTraits::IsVectorTypedData<T>::value>::type *enabler = nullptr ) const
+	{
+		const T *sourceDataCast = runTimeCast<const T>( m_sourceData );
+		const T *tweakDataCast = runTimeCast<const T>( m_tweakData );
+
+		const auto &currentElements = sourceDataCast->readable();
+		const auto &newElements = tweakDataCast->readable();
+
+		switch( m_mode )
+		{
+			case TweakPlug::ListAppend :
+				data->writable() = typename T::ValueType( currentElements );
+				for( const auto &element : newElements )
+				{
+					auto it = std::find( data->writable().begin(), data->writable().end(), element );
+					if( it != data->writable().end() )
+					{
+						std::rotate( it, it + 1, data->writable().end() );
+					}
+					else
+					{
+						data->writable().push_back( element );
+					}
+				}
```

Comment:
What do we think the right behaviour should be if the new element exists multiple times already? I think maybe we should remove all of them? I'm wondering if the most straightforward implementation might be to `remove_if()` all the new elements regardless of mode, and then append/prepend as needed?